### PR TITLE
BaSP-TG-16: List and methods for Get and Delete for tasks Made

### PR DESCRIPTION
**Pull request by FedeBrus64**
Get and Delete methods for tasks have been made. Styles and links also coded

### CODE REVIEW

**Review by Evelyn-Heredia — APPROVED** [review]
PR approved! It looks great!

**Review by FranGutierrez — APPROVED** [review]
Aproved by Tech Leader

File: src/Components/Tasks/index.js [hunk 1]
```diff
@@ -1,9 +1,76 @@
...
+  };
+
+  const deleteItem = (_id) => {
+    setTasksList([...tasksList.filter((task) => task._id !== _id)]);
+    window.alert('Task successfully deleted');
+  };
+
+  const addTask = ({ employee, project, title, description, date, done }) => {
+    const newItem = {
+      _id: Math.floor(Math.random() * 1000),
```
Comment (lucasrendo) [inline src/Components/Tasks/index.js:38]: This is unnecesary and can create a problem

File: src/Components/Tasks/List/list.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            <th id="project">Project</th>
+            <th id="title">Title</th>
+            <th id="description">Description</th>
+            <th id="date">Date</th>
+            <th id="done">Done</th>
+          </tr>
+        </thead>
+        <tbody>
+          {list.map((item) => (
+            <ListItem key={item._id} listItem={item} deleteItem={deleteItem} />
```
Comment (lucasrendo) [inline src/Components/Tasks/List/list.jsx:22]: editTask is a prop imported in listItem but not passed to the component or recieved from index here

**Review by lucasrendo — COMMENTED** [review]
Edit task button is not working correctly. It should be set so that is ready to use when it's integrated with the form

File: src/Components/Tasks/index.js [hunk 3]
```diff
@@ -1,12 +1,27 @@
...
   const [screen, changeScreen] = useState(false);
+  const [showingList, toggleList] = useState(true);
   const [formMethod, setMethod] = useState('POST');
   const [updTaskId, setId] = useState('');
 
+  useEffect(async () => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/tasks`);
+      const data = await response.json();
+      // eslint-disable-next-line no-console
```
Comment (BaltazarCortina1) [inline src/Components/Tasks/index.js:17]: Este comment no es neceasario

File: src/Components/Tasks/index.js [hunk 4]
```diff
@@ -1,12 +1,27 @@
...
+      setTasksList(data.data);
```
Comment (BaltazarCortina1) [inline src/Components/Tasks/index.js:18]: Aca deberias validar que la respuesta del back es exitosa antes de guardar la data

File: src/Components/Tasks/index.js [hunk 5]
```diff
@@ -33,15 +53,22 @@ function Tasks() {
...
       {screen ? (
         <Form formMethod={formMethod} back={() => backToList()} id={updTaskId} />
+      ) : showingList ? (
+        <List
+          list={tasksList}
+          setList={setTasksList}
+          deleteItem={deleteItem}
+          setMethod={'GET'}
+          editTask={() => editTask()}
+          setId={() => setId()}
```
Comment (BaltazarCortina1) [inline src/Components/Tasks/index.js:68]: Esto es lo mismo que pasar solo la funcion como callback:
`editTask={editTask}`

File: src/Components/Tasks/ListItem/ListItem.jsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    const options = {
+      method: 'DELETE',
+      headers: {
+        'Content-type': 'application/json'
+      }
+    };
+    fetch(`${process.env.REACT_APP_API_URL}/tasks/${listItem._id}`, options).then((response) => {
+      if (response.status !== 200) {
+        return response.json().then(({ message }) => {
+          throw new Error(message);
```
Comment (BaltazarCortina1) [inline src/Components/Tasks/ListItem/ListItem.jsx:15]: Esta bueno que valides si la request fue exitosa. Pero no hay ningun catch que agarre el error, si el status no es 200 te explota la app

File: src/Components/Tasks/ListItem/ListItem.jsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import Styles from './listItem.component.css';
```
Comment (BaltazarCortina1) [inline src/Components/Tasks/ListItem/ListItem.jsx:2]: Normalmente usamos variables que comienzan con mayusculas para los componentes. En este caso deberia ser `styles`

**Pull request merged**